File: src/person.cpp
#include "include/person.h"

big_int person::id = 0;

std::string person::get_first_name() const {
  return first_name_;
}

std::string person::get_second_name() const {
  return second_name_;
}

std::string person::get_address() const {
  return address_;
}

std::string person::get_passport_id() const {
  return passport_id_;
}

int person::get_money_limit() const {
  return money_limit_;
}

bool person::is_doubtful() const {
  return is_doubtful_;
}

big_int person::get_id() const{
  return id_;
}

void person::set_first_name(const std::string &first_name) {
  first_name_ = first_name;
}

void person::set_second_name(const std::string &second_name) {
  second_name_ = second_name;
}

void person::set_address(const std::string &address) {
  address_ = address;
}

void person::set_passport_id(const std::string &passport_id) {
  passport_id_ = passport_id;
}

void person::update() {
  is_doubtful_ = address_.empty() || passport_id_.empty();
  money_limit_ = (is_doubtful_ ? money_limit : 0);
}

void person::assign_id() {
  id_ = id++;
}

File: MIPT_bank_app1/MIPT_bank_app/app/src/main/java/com/example/mipt_bank_app/Person.java
package com.example.mipt_bank_app;


public class Person extends IPerson{
    public static int id;

    /*public Person() {
        if (id == 0) {
            std::ifstream file("max_person_id.txt");
            String s;
            file >> s;
            id = int(s);
            if (id != 0) ++id;
            file.close();
        }
    }*/

    @Override
    public String getFirstName() {
        return first_name_;
    }

    @Override
    public String getSecondName() {
        return second_name_;
    }

    @Override
    public String getAddress() {
        return address_;
    }

    @Override
    public String getPassportId() {
        return passport_id_;
    }

    @Override
    public int getMoneyLimit () {
        return money_limit_;
    }

    @Override
    public boolean isDoubtful () {
        return is_doubtful_;
    }

    @Override
    public int getId() {
        return id_;
    }


    public void setFirstName(String first_name) {
        first_name_ = first_name;
    }

    public void setSecondName(String second_name) {
        second_name_ = second_name;
    }

    public void setAddress(String address) {
        address_ = address;
    }

    public void setPassportId(String passport_id) {
        passport_id_ = passport_id;
    }

    public void assignId() {
        id = id + 1;
        id_ = id;
    }

    public void update() {
        is_doubtful_ = address_.isEmpty() || passport_id_.isEmpty();
        money_limit_ = (is_doubtful_ ? SupConstants.money_limit : 0);
    }
}
